### PR TITLE
Auth fix for regulation worker

### PR DESCRIPTION
**Pull request by saurav-malani**
**Fixes** # (*issue*)

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
## Checklist:
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added unit tests for the code
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: regulation-worker/internal/client/client.go [hunk 1]
```diff
@@ -37,12 +37,19 @@ func (j *JobAPI) Get(ctx context.Context) (model.Job, error) {
...
+
 	client := &http.Client{}
 	resp, err := client.Do(req)
 	if err != nil {
 		return model.Job{}, err
 	}
 	defer resp.Body.Close()
+	if err != nil {
+		fmt.Println(err)
+	}
```
Comment (lvrach) [inline regulation-worker/internal/client/client.go:52]: what do we error check here ?

File: regulation-worker/internal/client/client.go [hunk 2]
```diff
@@ -37,12 +37,19 @@ func (j *JobAPI) Get(ctx context.Context) (model.Job, error) {
 		return model.Job{}, err
 	}
 
+	workspaceToken := config.GetEnv("CONFIG_BACKEND_TOKEN", "22JnFdS3ZKDd0UfuEYowUeNi9fe")
```
Comment (lvrach) [inline regulation-worker/internal/client/client.go:40]: lets not hardcode the config.GetEnv here.

File: regulation-worker/cmd/main.go [hunk 3]
```diff
@@ -43,8 +43,8 @@ func Run(ctx context.Context) {
 	router := delete.NewRouter(&kvstore.KVDeleteManager{}, &batch.BatchManager{}, &apiManager)
 	svc := service.JobSvc{
 		API: &client.JobAPI{
-			WorkspaceID: config.GetEnv("workspaceID", "1001"),
-			URLPrefix:   config.GetEnv("urlPrefix", "https://api.rudderlabs.com:35359"),
+			WorkspaceID: config.GetEnv("workspaceID", "22JnFf8IlRAjIXtGUmhxylWMORi"),
+			URLPrefix:   config.GetEnv("urlPrefix", "https://leonidas-api.dev-rudder.rudderlabs.com"),
```
Comment (lvrach) [inline regulation-worker/cmd/main.go:47]: here would be a good place to pass WorkspaceToken.
But let us not use defaults, if WorkspaceID is not configured the service should not start

File: regulation-worker/internal/client/client.go (hunk 1 above)
Comment (saurav-malani) [reply regulation-worker/internal/client/client.go:52]: sorry, wanted to added `resp.Body.Close()` inside a defer func and check error, if not closed properly. Forgot to add `defer func`...added.

File: regulation-worker/internal/client/client.go (hunk 2 above)
Comment (saurav-malani) [reply regulation-worker/internal/client/client.go:40]: removed the `default` with `""`. And, panic if `workspaceToken` is empty.

File: regulation-worker/cmd/main.go (hunk 3 above)
Comment (saurav-malani) [reply regulation-worker/cmd/main.go]: done

**Review by lvrach — APPROVED** [review]
Approve, with minor comments

**Review by chandumlg — APPROVED** [review]

**Review by chandumlg — APPROVED** [review]
Tests failed. Need to fix them

**Pull request merged**